src/torque-offset-estimator.cpp

```diff
@@ -0,0 +1,240 @@
+      /* --- COMMANDS ---------------------------------------------------------- */
+      void TorqueOffsetEstimator::init(const std::string &urdfFile,
+                                       const Eigen::Matrix4d& m_torso_X_imu_,
+                                       const double& gyro_epsilon_)
+      {
+        try {
+          se3::urdf::buildModel(urdfFile,se3::JointModelFreeFlyer(),m_model);
+          // assert(m_model.nq == N_JOINTS+7);
+          // assert(m_model.nv == N_JOINTS+6);
+
+          jointTorqueOffsets.resize(m_model.nv-6);
+          jointTorqueOffsets.setZero();
+
+          ffIndex = 1;
+          torsoIndex = m_model.getJointId("torso_2_joint");
+        }
+        catch (const std::exception& e) 
+        { 
+          std::cout << e.what();
+          return SEND_MSG("Init failed: Could load URDF :" + urdfFile, MSG_TYPE_ERROR);
+        }
+        m_data = new se3::Data(m_model);
+        m_torso_X_imu.rotation() = m_torso_X_imu_.block<3,3>(0,0);
+        m_torso_X_imu.translation() = m_torso_X_imu_.block<3,1>(0,3);
+        gyro_epsilon = gyro_epsilon_;
+      }
+
+      void TorqueOffsetEstimator::computeOffset(const int& nIterations, const double& epsilon_)
+      {
+        if (sensor_offset_status == PRECOMPUTATION) {
+          SEND_MSG("Starting offset computation with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+          n_iterations = nIterations;
+          epsilon = epsilon_;
+          sensor_offset_status = INPROGRESS;
+        }
+        else if (sensor_offset_status == INPROGRESS) {
+          SEND_MSG("Collecting input signals. Please keep the graph running", MSG_TYPE_WARNING);
+        }
+        else { //sensor_offset_status == COMPUTED
+          SEND_MSG("Recomputing offset with no. iterations:"+ nIterations, MSG_TYPE_DEBUG);
+
+          // stdVecJointTorqueOffsets.clear();
+          current_progress = 0;
+          jointTorqueOffsets.setZero();
+
+          n_iterations = nIterations;
+          epsilon = epsilon_;
+          sensor_offset_status = INPROGRESS;
+        }
+        return;
+      }
+      
+      DEFINE_SIGNAL_INNER_FUNCTION(collectSensorData, dummy)
+      {
+        if (sensor_offset_status == INPROGRESS) {
+          // Check the current iteration status
+          int i = current_progress;
+          
+          if (i < n_iterations) {
+            SEND_MSG("Collecting signals for iteration no:" + i , MSG_TYPE_DEBUG_STREAM);
+
+            const Eigen::VectorXd& sot_enc = m_base6d_encodersSIN(iter);
+            const Eigen::VectorXd& IMU_acc = m_accelerometerSIN(iter);
+            const Eigen::VectorXd& tau = m_jointTorquesSIN(iter);
+
+            Eigen::VectorXd enc(m_model.nq);
+            joints_sot_to_urdf(sot_enc.tail(m_model.nv-6), enc.tail(m_model.nv-6));
+            //enc.tail(m_model.nv-6) = sot_enc.tail(m_model.nv-6);
+            base_sot_to_urdf(sot_enc.head<6>(), enc.head<7>());
```

Comment:
You don't have to do that because the first 6 values of base6d_encoders do not make sense (we don't have any sensor measuring the base state). Just set the first 7 values of enc to (0, 0, 0, 0, 0, 0, 1).